scenarios.md

```diff
@@ -230,6 +230,37 @@ A weakness is discovered in a widely used cryptographic algorithm and a decision
 1. Key revocation, chain of trust, etc. must all work for the expected lifetime of a version of the client software while these changes are made.
 1. The actions that different parties need to perform must be clearly articulated, along with the result of not performing those actions.
 
+### Scenario #10: Verifying Different Types of Artifacts
+
+Artifacts may be referenced in different ways and verification of signatures should handle the different types of artifact references:
+
+1. Immutable `sha256` digests
+1. Mutable tags
```

Comment:
Could you elaborate more on the immutable and mutable tags?

I am confused as the signed tags without digests are not secure since the tag can be dereferenced to any digests the server wishes.

Reply:
If a user pulls `alpine:edge` with notary checking enabled, this should work and verify the returned image was signed. Whether the check happens on the mutable tag, or only after the digest has been resolved from the tag, has not been specified here since implementations may differ.

Reply:
Whether the tag itself is immutable is a registry implementation feature. So this just acknowledges that tags could be either.

Reply:
@sudo-bmitch
The current notary v2 signature proposal signs the digest, which can be associated with a tag.
The scenario we outlined:
wabbit-networks signs the `net-monitor:v1` image with a digest of `sha256@121b221...`
Someone in acme-rockets deploys the `net-monitor:v1` image, referencing the tag. However, they have a policy that software must be signed with the `wabbit-networks` key. When the notary validation is applied, the tag is resolved to a digest. The notary v2 (nv2) client queries the registry for all signatures of the `sha256@121b221...` digest. As long as there's at least one signature from wabbit-networks for the given digest, it's considered a valid key.

The deployment could also reference the digest directly, skipping the above tag-->digest resolution. 

What I'm trying to understand is whether you're defining an additional scenario or a clarification to the existing [Notary v2 scenarios](https://github.com/notaryproject/requirements/blob/main/scenarios.md)

Reply:
@SteveLasker the existing implementations that only provide server side handling of the `sha256` digest would still be valid with this change. Those implementations could resolve the tag on the client side and have a path for the user to use an image with a tag and validate that the image was signed. Whether we want to also validate a signature on the tag itself was left to #43 .

I added this one after thinking about Notary v1, where we only signed tags, and therefore provided no verification for users that ran images with a `sha256` digest. I want to be sure that solutions we consider have a method to verify the image whether the client verifies a tag, sha, multi-platform manifest, Helm chart, or anything else we want to sign on the registry.